### PR TITLE
Add script, terraform and helm changes for static peer ids

### PR DESCRIPTION
**Pull request by AryehAlexander**
Thank you for contributing to Mina! Please see `CONTRIBUTING.md` if you haven't
yet. In that doc, there are more details around how to start our CI.

Explain your changes here.

Explain how you tested your changes here.

Checklist:

- [ ] Document code purpose, how to use it
  - Mention expected invariants, implicit constraints
- [ ] Tests were added for the new behavior
  - Document test purpose, significance of failures
  - Test names should reflect their purpose
- [ ] All tests pass (CI will check this if you didn't)
- [ ] Serialized types are in stable-versioned modules
- [ ] Does this close issues? List them:

Closes #0000
Closes #0000

### CODE REVIEW

File: automation/scripts/generate-keys-and-ledger.sh [hunk 1]
```diff
@@ -170,13 +176,29 @@ else
...
+
 fi
 echo "Online Fish Keyset:"
 cat keys/keysets/${TESTNET}_online-fish
 echo "Offline Fish Keyset:"
 cat keys/keysets/${TESTNET}_offline-fish
 echo
 
+
+if [[ -s "keys/libp2p/${TESTNET}/whale-block-producer-1" ]]; then
```
Comment (0x0I) [inline automation/scripts/generate-keys-and-ledger.sh:190]: Hmmm not sure I get this check...

File: automation/terraform/modules/kubernetes/testnet/helm.tf [hunk 2]
```diff
@@ -13,6 +13,14 @@ data "local_file" "genesis_ledger" {
   ]
 }
 
+data "local_file" "libp2p_peers" {
+  for_each = toset([ for config in var.block_producer_configs : config.name ])
+  filename = "../../../keys/libp2p/${var.testnet_name}/${each.key}"
```
Comment (0x0I) [inline automation/terraform/modules/kubernetes/testnet/helm.tf:18]: Probably worth prepending the filename path with one of Terraform's path vars (https://www.terraform.io/docs/configuration-0-11/interpolation.html#path-information) to be more explicit about where the keys should be stored. Also, there's an `artifact_path` VAR that was added that you may be able to use.

File: helm/block-producer/Chart.yaml [hunk 3]
```diff
@@ -2,7 +2,7 @@ apiVersion: v2
 name: block-producer
 description: A Helm chart for Mina Protocol's block producing network nodes
 type: application
-version: 0.4.6
+version: 0.5.0
 appVersion: 1.16.0
 annotations:
   artifacthub.io/changes: |
```
Comment (0x0I) [inline helm/block-producer/Chart.yaml:8]: hey, we need to update these changes and run `make changelog` to go along with the update.

File: automation/scripts/generate-keys-and-ledger.sh (hunk 1 above)
Comment (AryehAlexander) [reply automation/scripts/generate-keys-and-ledger.sh:190]: This is something similar we've done in other parts of the script as a quick way to (hopefully check) if we've already done work so we dont overwrite anything

File: helm/block-producer/Chart.yaml (hunk 3 above)
Comment (AryehAlexander) [reply helm/block-producer/Chart.yaml:8]: :+1: ran make changelog with new annotation

**Review by lk86 — APPROVED** [review]

**Comment by es92** [comment]
!approved-for-mainnet

**Pull request merged**